packages/playwright-core/src/client/browserType.ts

```diff
@@ -69,10 +63,11 @@ export class BrowserType extends ChannelOwner<channels.BrowserTypeChannel> imple
     assert(!(options as any).userDataDir, 'userDataDir option is not supported in `browserType.launch`. Use `browserType.launchPersistentContext` instead');
     assert(!(options as any).port, 'Cannot specify a port without launching as a server.');
 
-    const logger = options.logger || this._defaultLaunchOptions?.logger;
-    options = { ...this._defaultLaunchOptions, ...options };
+    const logger = options.logger || this._playwright._defaultLaunchOptions?.logger;
+    options = { ...this._playwright._defaultLaunchOptions, ...options };
     const launchOptions: channels.BrowserTypeLaunchParams = {
       ...options,
+      tracesDir: options.tracesDir ?? this._playwright._defaultTracesDir,
```

Comment:
I think it's better update `options` variable at line 67 with the new `tracesDir` instead of changing `launchOptions`, because `options` is later used at line 77 and so we want all the values there be correct.

Reply:
Since I moved `tracesDir` back into `_defaultLaunchOptions`, things should be good now.